backends/bmv2/psa_switch/midend.cpp

```diff
@@ -111,6 +112,9 @@ PsaSwitchMidEnd::PsaSwitchMidEnd(CompilerOptions &options, std::ostream *outStre
     if (BMV2::PsaSwitchContext::get().options().loadIRFromJson == false) {
         addPasses({
             options.ndebug ? new P4::RemoveAssertAssume(&typeMap) : nullptr,
+            new P4::TypeChecking(&refMap, &typeMap),
+            new SimplifyExternMethodCalls(&typeMap),
+            new P4::TypeChecking(&refMap, &typeMap),
```

Comment:
I find it weird that in `pna_nic` and `simple_switch` the second type checker is not there, but in `psa_switch` it is.